### PR TITLE
chore: avoid theme flashes

### PR DESCRIPTION
**Pull request by harish-sethuraman**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)
There are theme flashes in default and this code is not necessary at all 

- [ ] Documentation update
- [x] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
- [ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
- [ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
- [ ] Add autofix to a rule
- [ ] Add a CLI option
- [ ] Add something to the core
- [ ] Other, please explain:

#### What changes did you make? (Give an overview)

Removed unwanted code that causes flashing (similar to https://github.com/eslint/new.eslint.org/pull/196/files)

#### Is there anything you'd like reviewers to focus on?

Once check if flashing occurs with this preview build

### CODE REVIEW

File: docs/src/assets/js/themes.js [hunk 1]
```diff
@@ -8,11 +8,6 @@
         btn.setAttribute("aria-pressed", "false");
     }
 
-
-    let theme = window.localStorage.getItem("theme");
-    document.documentElement.setAttribute('data-theme', theme);
-    if (!theme) document.documentElement.setAttribute('data-theme', 'light');
-
```
Comment (mdjermanovic) [inline docs/src/assets/js/themes.js:15]: We are already setting the theme [here](https://github.com/eslint/eslint/blob/10249ad1a961463b6b347be71c074951ab8e2652/docs/src/_includes/layouts/base.njk#L48-L56) so setting it again in this script does seem unnecessary (and this code wasn't checking `prefers-color-scheme`).

But, we have some styles for [`html[data-theme="light"]`](https://github.com/eslint/eslint/blob/10249ad1a961463b6b347be71c074951ab8e2652/docs/src/assets/scss/tokens/themes.scss#L122), and now we're removing the code that sets it as the default, so maybe we should update [this code](https://github.com/eslint/eslint/blob/10249ad1a961463b6b347be71c074951ab8e2652/docs/src/_includes/layouts/base.njk#L48-L56) to set `"light"` as the default, like in the script on new.eslint.org ([here](https://github.com/eslint/new.eslint.org/blob/d1ff91e44fcd7814b355080a94d3545d2cad1ed0/src/_includes/layouts/base.html#L67-L75)), or set is direcly on the html tag (which could be better for users who have disabled JavaScript).

Comment (amareshsm) [reply docs/src/assets/js/themes.js:15]: https://github.com/eslint/eslint/blob/10249ad1a961463b6b347be71c074951ab8e2652/docs/src/assets/js/themes.js#L25-L28

We can also remove [these checks](https://github.com/eslint/eslint/blob/10249ad1a961463b6b347be71c074951ab8e2652/docs/src/assets/js/themes.js#L25-L28), right?

Comment (harish-sethuraman) [reply docs/src/assets/js/themes.js:15]: Looking into them today! :)

**Comment by harish-sethuraman** [comment]
bumping for review :)

File: docs/package.json [hunk 2]
```diff
@@ -10,12 +10,12 @@
...
         "images": "imagemin '_site/assets/images' --out-dir='_site/assets/images'",
-        "watch:sass": "sass --watch --poll src/assets/scss:src/assets/css",
+        "watch:sass": "sass --watch src/assets/scss:src/assets/css",
         "watch:eleventy": "eleventy --serve --port=2023",
         "build:sass": "sass --style=compressed src/assets/scss:src/assets/css --no-source-map",
         "build:eleventy": "npx @11ty/eleventy",
-        "start": "npm-run-all build:sass --parallel watch:*  --parallel images",
-        "build": "npm-run-all build:sass --parallel build:*   --parallel images"
+        "start": "npm-run-all --parallel watch:* images",
+        "build": "npm-run-all --parallel build:* images"
```
Comment (mdjermanovic) [inline docs/package.json:18]: Can you revert package.json? I think these changes are unrelated to this issue, so we could evaluate them separately from this PR.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!
